### PR TITLE
Add Italian Localization

### PR DESCRIPTION
**Pull request by htes9**
Let's use this clean one.

### CODE REVIEW

**Comment by Kamuno** [comment]
Have you tested the layouts with the different string lengths?

**Comment by htes9** [comment]
Settings layout is fine, I reduced one string that was out to make it better:
unknown

I'm working on making those strings shorten, keeping the same meaning:

unknown

File: app/src/main/res/values-it/strings.xml [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="app_name_long">Privacy Friendly QR Scanner</string>
+    <string name="version_number">Versione v%s</string>
+    <string name="delete_items">Cancella elementi</string>
+
+    <string name="scan_text">Posiziona questa finestra sopra il codice che vuoi scansionare.</string>
+    <string name="action_clear">Cancella Cronologia</string>
+    <string name="cannot_clear">La cronologia è già vuota</string>
+    <string name="knowRisks">Ho controllato il collegamento e voglio visitare il sito.</string>
+    <string name="domain">Dominio</string>
+    <string name="urlExplanation">I codici QR possono contenere collegamenti, che conducono a falsi siti web, capaci di compromettere i tuoi dati personali (siti di phishing) o trasferire malware (ad esempio virus) al tuo dispositivo.\n\nCome conseguenza dovresti <b>controllare</b> se il seguente collegamento ti sembra corretto:</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:59]: I codici QR possono contenere link che conducono a.....

Then, every other "collegamento" -> link.

File: app/src/main/res/values-it/strings.xml [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="further_info"><a href="https://www.secuso.informatik.tu-darmstadt.de/de/research/results/nophish-flyer-deutsch/unterstuetzung-bei-der-erkennung-von-phishing-angriffen/">Informazione</a> come controllare i collegamenti forniti dai colleghi della TU Darmstadt. Se non ti fidi del collegamento, clicca non mi fido &amp; non salvare.</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:60]: Same a before: not "collegamento", but link

File: app/src/main/res/values-it/strings.xml [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="delete_history">Vuoi cancellare la cronologia?</string>
+    <string name="ok">Okay</string>
+    <string name="element_removed">Elemento rimosso.</string>
+    <string name="conform_url">Per favore, conferma che hai controllato il collegamento.</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:64]: Per favore, conferma di aver controllato il link

File: app/src/main/res/values-it/strings.xml [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="wifi_explanation">Questa versione dell\'app non può connettersi a reti WIFI automaticamente a causa dei permessi necessari. Qui sotto trovi alcuni dettagli sulla rete.</string>
+    <string name="encryption">Criptazione</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:66]: Crittografia

File: app/src/main/res/values-it/strings.xml [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="title_activity_settings">Impostazioni</string>
+    <string name="title_activity_help">Aiuto</string>
+    <string name="title_activity_about">Al riguardo</string>
+    <string name="Settings">Impostazioni</string>
+    <string name="QRScanner">QR Scanner</string>
+    <string name="QRGenerator">QR Generator</string>
+    <string name="History">Cronologia</string>
+
+    <string name="slide1_heading">Benvenuto!</string>
+    <string name="slide1_text">Benvenuto su privacy friendly QR scanner. Questo tutorial ti illustrerà le cose più importanti che devi conoscere.</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:27]: PRIVACY FRIENDLY

File: app/src/main/res/values-it/strings.xml [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="title_activity_result_email">Indirizzo E-mail trovato</string>
+    <string name="title_activity_result_send_email">E-mail trovata</string>
+    <string name="title_activity_result_product">Prodotto trovato</string>
+    <string name="title_activity_result_tel">Numero di telefono trovato</string>
+    <string name="title_activity_result_wifi">WiFi trovata</string>
+    <string name="title_activity_result_contact">Contatto trovato</string>
+    <string name="action_help">Aiuto</string>
+
+    <!-- ### WELCOME DIALOG ### -->
+    <string name="welcome_text">L\'app PRIVACY FRIENDLY QR Code Scanner consente di scansionare codici a barre e codici-QR di varia tipologia richiedendo un numero minimo di permessi e rilevando collegamenti dannosi.</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:171]: link, not "collegamenti"

File: app/src/main/res/values-it/strings.xml [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+    <string name="welcome">Benvenuto</string>
+    <string name="viewhelp">Visualizza Aiuto</string>
+
+    <!--<string name="help_privacy_heading">Privacy Info</string>
+    <string name="about_author_contributors">Contributors</string>
+    <string name="privacy_friendly">This application belongs to the group of PRIVACY FRIENDLY Apps developed by Technische Universität Darmstadt. Sourcecode licensed under GPLv3. Images copyright TU Darmstadt and Google Inc.</string>
+    <string name="more_info">More information can be found on:</string>
+    <string name="github" translatable="false"><a href="https://github.com/SecUSo/privacy-friendly-qr-scanner">Github-Repo</a></string>-->
+    <string name="help_permission1_description">L\'applicazione richiede il permesso di accedere alla fotocamera, che viene utilizzato per scansionare i codici a barre ed i codici QR attraverso la fotocamera.</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:180]: che viene utilizzata

File: app/src/main/res/values-it/strings.xml [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+
+
+    <string name="help_whatis">Che cos\'è PRIVACY FRIENDLY QR Scanner?</string>
+
+    <string name="help_usability">La scansione non funziona, cosa posso fare?</string>
+    <string name="help_usability_answer">Se non puoi scansionare un codice, aumenta la distanza tra lo scanner ed il codice.</string>
+    <string name="help_privacy">Cosa rende l\'app privacy-friendly?</string>
+    <string name="help_privacy_answer">L\'app Privacy Friendly non richiede alcun permesso. Inoltre non include alcun meccanismo di tracking od annuncio.</string>
+    <string name="help_permission">Quali permessi sono richiesti da PRIVACY FRIENDLY QR Scanner?</string>
+    <string name="help_permission_answer">L\'applicazione richiede il permesso di accedere alla fotocamera, che viene utilizzato per scansionare i codici a barre ed i codici QR attraverso la fotocamera.</string>
```
Comment (corvinusWiz) [inline app/src/main/res/values-it/strings.xml:199]: utilizzata

File: app/src/main/res/values-it/strings.xml (hunk 6 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml:171]: done

File: app/src/main/res/values-it/strings.xml (hunk 4 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml]: done

File: app/src/main/res/values-it/strings.xml (hunk 3 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml]: done

File: app/src/main/res/values-it/strings.xml (hunk 2 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml]: done

File: app/src/main/res/values-it/strings.xml (hunk 1 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml]: done

File: app/src/main/res/values-it/strings.xml (hunk 7 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml:180]: "che viene utilizzato", we are talking about the **permission**, hence, "utilizzato"

Do you agree?

File: app/src/main/res/values-it/strings.xml (hunk 8 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml:199]: "che viene utilizzato", we are talking about the permission, hence, "utilizzato"

Do you agree?

Comment (htes9) [reply app/src/main/res/values-it/strings.xml:199]: hold on, i know how to fix it

Comment (htes9) [reply app/src/main/res/values-it/strings.xml:199]: ```L\'applicazione richiede il permesso di accedere alla fotocamera, utilizzata per scansionare i codici a barre ed i codici QR.```

File: app/src/main/res/values-it/strings.xml (hunk 7 above)
Comment (htes9) [reply app/src/main/res/values-it/strings.xml:180]: ```L\'applicazione richiede il permesso di accedere alla fotocamera, utilizzata per scansionare i codici a barre ed i codici QR.```

File: app/src/main/res/values-it/strings.xml (hunk 8 above)
Comment (corvinusWiz) [reply app/src/main/res/values-it/strings.xml:199]: That would have been my proposal. perfect.

**Comment by htes9** [comment]
Let me know if you find something else to change: https://github.com/htes9/privacy-friendly-qr-scanner/blob/master/app/src/main/res/values-it/strings.xml

**Comment by htes9** [comment]
Out of curiosity I did a complete refactor to AndroidX and updated the localization, there is only a little bug that is beyond my grasp when you access History and no element was prior saved, I forked again because I had a bit of confusion. Let me know if you can reproduce the bug with AndroidX and what are your projects for full port. Thank you.

**Pull request closed without merging**